script.rb

```diff
@@ -0,0 +1,155 @@
+module Enumerable
+  def my_each
+    i = 0
+    while i < length
+      yield(self[i])
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    i = 0
+    while i < length
+      yield(self[i], i)
+      i += 1
+    end
+  end
+
+  def my_select
+    temp = []
+    i = 0
+    while i < length
+      temp << self[i] if yield(self[i])
+      i += 1
+    end
+    temp
+  end
+
+  def my_all?
+    i = 0
+    while i < length
+      return true if yield(self[i]) && yield(self[i + 1])
+
+      i += 1
+      return false
+    end
+  end
```

Comment:
- For my_all? handle the case when no block or argument is given, it should return true when none of the collection members are false or nil.

- For the built-in method
`[1, true, 'hi', []].all?` returns `true`
`[1, false, 'hi', []].all?` returns `false`
- Your method gives the error `LocalJumpError:
       no block given (yield)
`
- The error occurs because your methods are always expecting a block, however, there are cases when a block isn't provided.

Reply:
- For my_all, my_none and my_any let's handle cases when the argument is a REGEX or a CLASS.

- For the built-in method
`%w[ant bear cat].all?(/a/) => true`
`[1, 2i, 3.14].all?(Numeric) => true`

- Your methods produce the following error because they aren't expecting any arguments to be passed to them

> ArgumentError:
        wrong number of arguments (given 1, expected 0)

Reply:
my_all? my_any? my_none? methods updated